### DISCUSSION TITLE
Does duckplyr have support for/plans for "lazy operations" using Views?

### DISCUSSION THREAD

**Discussion by JanSulavik** [duckdblabs/duckplyr#652] [Q&A]
Hello,

first of all: this package looks really exciting, combining dplyr and Duck DB sounds like a really great idea.

I have a question: in Python, I use Polars' Lazy API (LazyFrames) to work with data too large for my RAM (mainly huge parquet files).

In R, I have been mostly using DuckDB's Views (virtual tables) for the same purpose, but then using SQL-syntax of course.

Does duckplyr have support for/plans for supporting use of dplyr syntax on Views?

I looked at duckplyr reference manual on CRAN and browsed through the Issues here, but I haven't found an explicit mention of Views yet.

Thank you very much in advance for answer and keep up the good work with duckplyr.

Best regards,
Jan

**Comment by krlmlr** [comment]
Thanks. With the released version, you can read directly from Parquet or CSV files using the `read_*_duckdb()` functions, or from an existing table in the (internal) database using `read_sql_duckdb()` . By and large, I think the package is now in a shape that supports this kind of workflow.
